### ISSUE TITLE
Uploads going to a "page not found" error

### ISSUE THREAD

**Issue by babaklar** [MESH-Research/knowledge-commons-works#656]
Some users are reporting that after they upload items to Works, they receive a "page not found" error.

https://works.hcommons.org/records/889ef-a6508
https://works.hcommons.org/records/j3ykp-ae494

I will add more links to this issue if we receive them.

**Comment by bjr70** [comment]
Claus Janew has a set of issues that I think are related.

Laws of Form https://works.hcommons.org/records/889ef-a6508
Awareness meditation template https://works.hcommons.org/records/540n4-neg52 

Context from his email: 
Laws of Form https://works.hcommons.org/records/889ef-a6508 has a new version. This version, even though I pressed "Publish", does not show up in the profile and also the form is not available from the dashboard https://works.hcommons.org/me/uploads?q=&l=list&p=1&s=10&sort=updated-desc ("page not found"). In the "Imprint" section, there is also an error displaying page numbers. This has been moved from Essay to Journal Article, which is ok since both are correct, but this may have triggered the bug.
Awareness meditation template https://works.hcommons.org/records/540n4-neg52 is saved as a draft (on your server), but is not available from the dashboard https://works.hcommons.org/me/uploads?q=&l=list&p=1&s=10&sort=updated-desc ("page not found").

**Comment by monotasker** [comment]
I was just able to manually publish https://works.hcommons.org/records/889ef-a6508 myself. The new version now appears in a general search, so it *should* appear generally. Why it didn't publish properly before is still a mystery.

**Comment by monotasker** [comment]
If the author of https://works.hcommons.org/records/j3ykp-ae494 intends for it to be published, we can manually publish it too. I can do it, or any of us with admin access can. (Search for it from administration > drafts, click through to the upload form, and click "publish" on the last tab.) This isn't a fix for the problem, but it at least will get these records published

**Comment by monotasker** [comment]
Oh, also, for https://works.hcommons.org/records/889ef-a6508, if the author edits the metadata they can change the publication type back to "Book Section" and then they can fix the page numbers. (The page numbers were placed in "total book pages" instead of "section pages". These fields don't appear at all when the item is switched to "Journal article" as the resource type.)

**Comment by monotasker** [comment]
The link for https://works.hcommons.org/records/540n4-neg52 works if you add "?preview=1" on the end, since it's still in draft state (so its a record preview).

**Comment by monotasker** [comment]
Ooooh, there's a regression in the user dashboard record list. The links for draft records should have "?preview=1" on the end. There should also be distinct "edit" and "preview" buttons. So that issue (not being able to view drafts from dashboard) is going to happen for everyone. I'll prioritize fixing it tomorrow. In the meantime, adding "?preview=1" to any link for a draft should work.

**Comment by babaklar** [comment]
@monotasker For this record, [https://works.hcommons.org/records/j3ykp-ae494](https://works.hcommons.org/records/j3ykp-ae494), the issue is the user doesn't want to publish it **yet**. He was getting the metadata ready without adding a file until he is 100% ready--to quote him directly "i was just trying to get it ready to publish (it would be ideal if i could set it up, get the doi, so we can then add that to the frontmatter of the book) and have it be restricted for about a week (while we order a print proof) and then open it to the public once we have an approved book…" 

So I don't want to publish it yet because there is no file added. But since he can't access the record due to the "page not found" error. However, if a fix is coming relatively soon (within the next week or so) I can tell him we're working to get the draft record restored so he can access it and add the file once he's ready. Does that make sense?

And again, this is not urgent! :)

**Comment by bjr70** [comment]
Messaged Claus. Will let you know if he wants that deposit published.

**Comment by monotasker** [comment]
If you're talking about https://works.hcommons.org/records/889ef-a6508, I already published it since he had hit "publish" himself (see above). If you mean https://works.hcommons.org/records/540n4-neg52 (the draft) he should now be able to view and publish it himself, unless he runs into problems with that record too. He can (again) view it now if he adds "?preview=1" to the url.

**Comment by bjr70** [comment]
> If you're talking about https://works.hcommons.org/records/889ef-a6508, I already published it since he had hit "publish" himself (see above). If you mean https://works.hcommons.org/records/540n4-neg52 (the draft) he should now be able to view and publish it himself, unless he runs into problems with that record too. He can (again) view it now if he adds "?preview=1" to the url.

Yup, I emailed him. Will let you know. :)

**Comment by monotasker** [comment]
I think we can close this one. The issues with drafts being visible are now fixed. I'm not sure why he wasn't able to publish that one record's new version himself, and I haven't been able to reproduce that behaviour (with a non-admin user). So now that we manually published it for him, I think we should just wait to see whether we get any other similar reports of failed publication for new versions.
